src/ADAL.PCL/AcquireTokenNonInteractiveHandler.cs

```diff
@@ -111,11 +113,15 @@ public AcquireTokenNonInteractiveHandler(RequestData requestData, UserAssertion
                     {
                         throw new AdalException(AdalError.MissingFederationMetadataUrl);
                     }
+                    if (string.IsNullOrWhiteSpace(userRealmResponse.CloudAudienceUrn))
```

Comment:
this if check and defaultAppliesTo variable should be inside the WsTrustRequest class only. Let the request class make the decision on how the value is used.